File: front/src/app/core/commands/ResetPasswordCommand.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { ServerMessage, ServerMessageService } from '../services/ServerMessageService';
import { filter, catchError, switchMap } from 'rxjs/operators';
import { Product } from '../models/Product';
import { CommandHelper } from './CommandHelper';

@Injectable({
  providedIn: 'root'
})
export class ResetPasswordCommand {
  constructor(private httpClient: HttpClient, private commandHelper: CommandHelper) {
  }

  execute(resetCode: string, newPassword: string, email: string): Observable<ServerMessage>{
    const url = '/api/resetPassword';
    return this.commandHelper.getResponseStatusHandler(this.httpClient.post(url, {resetCode, newPassword, email}), true);
  }
}

File: backend/src/Users.Application/Commands/ResetPassword/ResetPasswordCommand.cs
using Common.Application.Commands;
using System.ComponentModel.DataAnnotations;
using Users.Domain.Auth;

namespace Core.Command.Commands.ResetPassword
{
    public class ResetPasswordCommand : ICommand
    {
        [Required] public string NewPassword { get; }
        [Required] public ResetCode ResetCode { get; }
        [Required] public string Email { get; }

        public ResetPasswordCommand(string newPassword, ResetCode resetCode, string email)
        {
            NewPassword = newPassword;
            ResetCode = resetCode;
            Email = email;
        }
    }
}